openff/toolkit/topology/molecule.py

```diff
@@ -6468,14 +6468,40 @@ def visualize(self, backend="rdkit", width=500, height=300):
 
         raise ValueError("Could not find an appropriate backend")
 
-    def perceive_residues(self):
+    def perceive_residues(self, substructure_file_path=None, strict_chirality=True):
         """
         Perceive residue substructure and fill atoms metadata accordingly.
+
+        Perceives residues by matching substructures in the current molecule with a substructure dictionary file,
+        using SMARTS.
+
+        Parameters
+        ----------
+        substructure_file_path : str, optional, default=None
+            Path to substructure library file in JSON format. Defaults to using built-in substructure file.
+        strict_chirality: bool, optional, default=True
+            Whether to use strict chirality symbols (stereomarks) for substructure matchings with SMARTS.
         """
         # Read substructure dictionary file
-        substructure_file_path = get_data_file_path('proteins/aa_residues_substructures.json')
+        if not substructure_file_path:
+            substructure_file_path = get_data_file_path('proteins/aa_residues_substructures.json')
         with open(substructure_file_path, 'r') as subfile:
             substructure_dictionary = json.load(subfile)
+
+        # TODO: Think of a better way to deal with no strict chirality case
+        # if ignoring strict chirality, remove/update keys in inner dictionary
+        if not strict_chirality:
+            # make a copy of substructure dict
+            substructure_dictionary_no_chirality = deepcopy(substructure_dictionary)
+            # Update inner key (SMARTS) maintaining its value
+            for res_name, inner_dict in substructure_dictionary.items():
+                for smarts, atom_types in inner_dict.items():
+                    smarts_no_chirality = smarts.replace('@', '')  # remove @ in smarts
+                    substructure_dictionary_no_chirality[res_name][smarts_no_chirality] = \
+                        substructure_dictionary_no_chirality[res_name].pop(smarts)  # update key
+            # replace with the new substructure dictionary
+            substructure_dictionary = substructure_dictionary_no_chirality
+
```

Comment:
It doesn't look like this information will change over multiple function calls to me. Perhaps you could make and read another file?

Reply:
As far as I see this, we might want sometimes to read with strict chirality and sometimes without it, so, I guess it is fine that this doesn't persist in multiple function calls. It is true that this will need to read the substructure dictionary every time but I think that's fine for now, that's fast and the memory footprint should be negligible. Maybe something to consider if we have a LOT of molecules, true.

In the long run we could be just fine by not using strict chirality at all, and making that the default. We would have to check this with detail.

Reply:
I agree with @ijpulidos -- We're still in "behavior seeking" mode right now with this new functionality, so it's probably best to keep this "messy" to make it easier to modify in later iterations.